sqrl-planner/src/main/java/com/datasqrl/engine/stream/flink/plan/SqrlToFlinkSqlGenerator.java

```diff
@@ -379,23 +380,22 @@ private SqlCreateTable registerSinkTable(WriteSink sink, RelNode relNode,
       EngineSink engineSink = (EngineSink) sink;
       TableConfig engineConfig = engineSink.getStage().getEngine().getSinkConfig(engineSink.getNameId());
 
-      Optional<StagePlan> stagePlan = stagePlans.stream()
+      //Add partitioning keys if engine supports them
+      StagePlan stagePlan = stagePlans.stream()
           .filter(f -> f.getStage() == engineSink.getStage())
-          .findFirst();
+          .findFirst().orElseThrow();
       TableConfigBuilder builder = engineConfig.toBuilder();
-      if (stagePlan.isPresent()) {
-        StagePlan stagePlan1 = stagePlan.get();
-        if (isPartitionable(stagePlan1.getStage().getName())) {
-          DatabaseStagePlan dbPlan = (DatabaseStagePlan) stagePlan1;
-          String tableId = engineSink.getNameId();
-          Optional<IndexDefinition> optIndex =  dbPlan.getIndexDefinitions().stream()
-              .filter(i -> i.getTableId().equals(tableId))
-              .findFirst();
-          if (optIndex.isPresent()) {
-            IndexDefinition mainIndex = optIndex.get();
-            builder.setPartitionKey(mainIndex.getColumnNames());
-          }
-
+      if (engineSink.getStage().supportsFeature(EngineFeature.PARTITIONING)) {
```

Comment:
👍

Reply:
I'm going to add a configuration switch for partition and primary keys so we can target all permutations of the iceberg connector. This code shouldn't change, we can move it to the engine.